File: sach/de3379.cpp
//DE3379.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
int n;
bool ktsnt(long long k) {
    if (k<2) return false;
    int ck = sqrt(k);
    for (int x=2; x<=ck; x++)
        if (k % x == 0) return false;
    return true;
}
int main()
{
    long long dem, y, s;
    cin>>n;
    dem=0;
    int cn=sqrt(n);
    for (int p=2; p<=cn; p++) {
        if (!ktsnt(p)) continue;
        y=p; s=1;
        while (y<=n) {
            s+=y;
            if (ktsnt(s)) dem++;
            y*=p;
        }
    }
    cout<<dem;
    return 0;
}

File: sach/de3379.py
//DE3379.PY - Tran Huu Nam - hsgtin.vn - 2024
import math
def is_prime(k):
    if k < 2:
        return False
    ck = int(math.sqrt(k))
    for x in range(2, ck + 1):
        if k % x == 0:
            return False
    return True
def main():
    n = int(input())
    dem = 0
    cn = int(math.sqrt(n))
    for p in range(2, cn + 1):
        if not is_prime(p):
            continue
        y = p
        s = 1
        while y <= n:
            s += y
            if is_prime(s):
                dem += 1
            y *= p
    print(dem)
if __name__ == "__main__":
    main()

